### DISCUSSION TITLE
Sheet of recently used files

### DISCUSSION THREAD

**Discussion by WizzardUU** [saulpw/visidata#1581] [Ideas]
It could be helpful for those, who often work with the same files for a while, to have a special sheet, that keeps a record of the recently opened files. 

The sheet should have columns like: path, filename, last opened / last saved, size. 
Similar to the current sheets sheet, you would load the file by navigating to the row and pressing Enter.

I imagine there are some restrictions on what sources or ways of loading could work with that, but anything file based would be a start.

Of course, for the first file of a vd instance this could be achieved by means of the OS, but it would be more convenient, if you're already working in a vd instance and need another file.

**Comment by geekscrapy** [comment]
I like this suggestion! I agree, `history` only gets you so far.
I might add that the path to the files would be absolute? I jump around multiple locations so a relative path would be no good.
I think the sheet should only show the root source sheet opened either at the terminal or via `o`?
Bonus points if the sheet had a (peek?!) column that provides the first row of data

**Comment by frosencrantz** [comment]
I like this suggestion, too.  You might need to include a column that provided options. like for loading such as delimiter for *sv files. I like geekscrapy's preview suggestion, too.

I was thinking along these lines recently.  I was coming from what should be the default sheet when VisiData starts?  Currently without any arguments it looks at the current directory.  But it would be useful if it could show recently opened files like you are suggesting. Or it could be a set of defined bookmarks.  It seemed like this would make for a good category of Visidata plugins, start sheet.  Web browsers have this category of plugin.

**Reply by WizzardUU** [reply]
Yes, this was actually my inspiration as well: The vim plugin "startify". But a shortcut to a sheet would be totally fine as well.
Since a sheet would be editable anyway, you could also insert bookmarks manually. Maybe have a column to set a flag to keep a row from being dropped due to moving down the list.

**Reply by frosencrantz** [reply]
Yes that seems like a great example of what such a plugin for VisiData could be.

Link: https://github.com/mhinz/vim-startify
